### ISSUE TITLE
New override location 

### ISSUE THREAD

**Issue by james2432** [AutoMcG/steamdeck_tools#28]
https://steamcommunity.com/games/1675200/announcements/detail/3301726014480099880

Custom boot videos

    Steam Deck will now look for custom boot and suspend animations in steamui/overrides/movies/
    . Please note that you'll need to create this folder first.
    Play custom boot animations in full screen

**Comment by Carleslc** [comment]
> Play custom boot animations in full screen

Does this mean that the CSS will be fixed now without modifying anything?

So it may work just setting the boot video in the new folder `/home/deck/.local/share/Steam/steamui/overrides/movies/` and rename the file to `deck_startup.webm`

`vidswap` script and randomizer should be updated to use this folder instead of the old one.

**Comment by james2432** [comment]
seems you just dump the deck_startup.webm (the name of the file you want to override) with new one, no more truncating, no more js to have it go full screen

**Comment by AutoMcG** [comment]
Yes, I plan on updating vidswap to take advantage of the new location with symlinks but it should continue to work in its current form (it will just continue overwrite what's there which will be replaced with the fresh version next update anyway).

One thing I'm not sure about is if this new change will also allow for >10 second videos - that's a modification to library.js and is not mentioned in the release notes yet. I'll be looking into testing that before I update for this issue. Shouldn't be difficult with the new vidswap_core.sh architecture.

**Comment by AutoMcG** [comment]
Also I only plan on merging this after the Steam Deck Client update hits primetime. Anyone who wants it before then can use the branch I'll create tonight for it.

**Comment by DRok17** [comment]
I soft-bricked my vita this morning making CSS and JS file changes. I suspect it was the JS file, as it was definitely different and didn't contain the code i,1e4 that needed to be replaced. Just a heads up.

**Comment by AutoMcG** [comment]
@DRok17 What JS file changes did you make? Just a sed replace on i,1e4 or did you also truncate with a static size saved somewhere in your code? (Or other?)

**Comment by DRok17** [comment]
Basically, I replaced a Preview JS file for a Stable one, I think that's where I screwed up. The sed command wouldn't have worked, as the code seems to be missing or has changed. I truncated the file based on what was originally put there by the OS, so the size should have matched.

**Comment by AutoMcG** [comment]
I have confirmed the limit is still 10 seconds without library.js changes. The matching pattern to find looks like it's now s,1e4. Might put some time into attempting to comprehend this file as it's not great to have to look for this specific string now that we have witnessed the letter might change. I can probably come up with a regex match to look for g.KS as the base to look for instead. 

library.css has been updated in the beta channel to already include the full screen modification as expected. Haven't done a full diff to look for other changes but it's nice to have that confirmed. 

Ran out of time to create a branch for this update but the good news is I confirmed any installed randomizer should continue to function after the update. The first boot after updating will be the standard steam boot, the one after that will be randomized.

**Comment by AutoMcG** [comment]
Also: Still need to test size constraint to see if there's a new limit or... what exactly. The only 30 second webm I have on-hand immediately is one I already compressed down.

**Comment by AutoMcG** [comment]
They updated it again on us. 

New override directory is `/home/deck/.steam/root/config/uioverrides/movies/`

**Comment by james2432** [comment]
was about to comment the same thing:

https://steamcommunity.com/games/1675200/announcements/detail/3301726014486735949

**Comment by AutoMcG** [comment]
28-new-override-location branch should now work on steam client beta with new overrides dir. 

Branch is slightly messy in current form, going to keep it on the shelf for now until override feature goes to stable. 
I had started laying the groundwork for library.js edits but didn't fully implement it, will probably wait at this point. 

Sidenote: I used a 30 MB .webm with the new overrides and it worked just fine and dandy.
